torchtune/models/llama3/_tokenizer.py

```diff
@@ -83,11 +100,18 @@ def _validate_special_tokens(
         self,
     ):
         """
-        Validate that required special tokens are passed into the tokenizer. The
-        following special tokens are required: <|begin_of_text|>, <|end_of_text|>,
-        <|start_header_id|>, <|end_header_id|>, <|eot_id|>, <|eom_id|>, <|python_tag|>
+        Validate that required special tokens are passed into the tokenizer.
         """
-        for token in LLAMA3_SPECIAL_TOKENS.keys():
+        for token in [
+            "<|begin_of_text|>",
+            "<|end_of_text|>",
+            "<|start_header_id|>",
+            "<|end_header_id|>",
+            "<|eom_id|>",
+            "<|eot_id|>",
+            "<|python_tag|>",
+            "<|image|>",
+        ]:
```

Comment:
Isn't the whole point of this tokenizer design that we can use different builders to support different sets of special tokens? It feels weird that we would then assert on tokens that may not be present in all llama3 tokenizers

Reply:
as discussed offline, all but the image token is required. so will remove that one